utilities/random_number_generation/bbs.py

```diff
@@ -0,0 +1,42 @@
+from primes import getStrongBlumPrime
+
+"""
+    An implementation of the Blum Blum Shub pseduo-random 
+    number generator.
+    
+    General Notes:
+        - BBS must be initialized with a random seed
+        - In order for BBS to produce cryptographically secure
+          output, the seed must be truly random
+        - In order for the generator to be practically secure, the 
+          bit length of the modulus should be no less than 2048
+        - The functions used to generate the modulus of the
+          function rely on PyCryptodome's Crypto.Util package
+"""
+
+"""
+    Returns an integer n = p * q where p and q are primes of the
+    form needed for BBS.
+"""
+def getModulus(bits=2048):
+    psize = bits // 2
+    return getStrongBlumPrime(psize) * getStrongBlumPrime(psize)
```

Comment:
What about `getStrongBlumPrime` makes it so that you can't just do `getStrongBlumPrime**2`? Especially since you're passing in the same param `psize` twice.

Reply:
I wondered the same thing at first but then I realized something very subtle. If we do "getStrongBlumPrime**2" it squares the result form one function call: (getStrongBlumPrime)^2 = x^2. We need both factors to be different though so since each function call returns a different random value, getStrongBlumPrime * getStrongBlumPrime = x * y.